src/main/java/com/gluonhq/richtext/viewmodel/RichTextAreaViewModel.java

```diff
@@ -474,4 +511,31 @@ private void updateProperties() {
         undoStackEmptyProperty.set(commandManager.isUndoStackEmpty());
         redoStackEmptyProperty.set(commandManager.isRedoStackEmpty());
     }
+
+    public FaceModel getCurrentFaceModel() {
+        return new FaceModel(getTextBuffer().getText(), getTextBuffer().getDecorationModelList(), getCaretPosition());
+    }
+
+    void newFaceModel() {
+        Platform.runLater(() -> {
+            setFaceModel(EMPTY_FACE_MODEL);
+            setFaceModel(new FaceModel());
```

Comment:
Why do we need to set the FaceModel twice?

Reply:
Control-skin-viewModel are connected via properties and listeners. Actions go to the VM. In order to trigger a change in skin or in the control we need to invalidate the properties in the VM.
For that, instead of null I use this empty FM.
Imagine this: open app (new FM is set in control and VM. Add a letter, don’t save. Press New again. Since FM in control still has the initial FM there wouldn’t be a difference and the skin wouldn’t clean the content.
Does this make sense?

Reply:
I was checking the code and it seems that we can set `EMPTY_FACE_MODEL` as default value on both Control and ViewModel's `faceModelProperty` to get rid of this double assignment. Thoughts?